e2e/tests/uniswap_trade_test.rs

```diff
@@ -0,0 +1,186 @@
+use contracts::{ERC20Mintable, GPv2Settlement, UniswapV2Factory, UniswapV2Router02};
+use ethcontract::prelude::{Account, Address, Http, PrivateKey, Web3, U256};
+use model::{DomainSeparator, OrderCreationBuilder, OrderKind};
+use orderbook::orderbook::OrderBook;
+use secp256k1::SecretKey;
+use serde_json::json;
+use std::{str::FromStr, sync::Arc};
+use web3::signing::SecretKeyRef;
+
+const TRADER_A_PK: [u8; 32] = [
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1,
+];
+const TRADER_B_PK: [u8; 32] = [
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 2,
+];
```

Comment:
If you prefer you can use the `hex!` macro here that we have in other parts of the code.